mux_test.go

```diff
@@ -1982,3 +1982,54 @@ func TestNoMatchMethodErrorHandler(t *testing.T) {
 		t.Error("Should not have any matching error. Found:", match.MatchErr)
 	}
 }
+
+func TestErrMatchNotFound(t *testing.T) {
+	emptyHandler := func(w http.ResponseWriter, r *http.Request) {}
+
+	r := NewRouter()
+	r.HandleFunc("/", emptyHandler)
+	s := r.PathPrefix("/sub/").Subrouter()
+	s.HandleFunc("/", emptyHandler)
+
+	// Regular 404 not found
+	req, _ := http.NewRequest("GET", "/sub/whatever", nil)
+	match := new(RouteMatch)
+	matched := r.Match(req, match)
+
+	if matched {
+		t.Error("Subrouter should not have matched that")
```

Comment:
I would add some context - `”subrouter should not have matched: got %v”, route”` to aid debugging.